src/main/scala/org/broadinstitute/hail/driver/ExportVCF.scala

```diff
@@ -50,6 +50,17 @@ object ExportVCF extends Command {
     case _ => "String"
   }
 
+  def appendOption(sb: StringBuilder, toAppend: Option[Array[Int]]): Unit = {
+    if (toAppend.isDefined)
+      toAppend.foreach(_.foreachBetween(sb.append(_))(sb += ','))
+    else
+      sb += '.'
+  }
+
+  def append[@specialized(Int, Double) T <: AnyVal](sb: StringBuilder, toAppend: Array[T]): Unit = {
+    toAppend.foreachBetween(sb.append(_))(sb += ',')
```

Comment:
Sorry it took me so long to get feedback.  We've been putting out some fires.

So I don't think `@specialized` changes the semantics, just the performance/boxing behavior for primitive types.  So overloading resolution is still based on the type coming from the type bound, namely, the version of `append` that takes an any and calls `toString`.

Specifically:

```
scala> object A { def f(i: Int) = println("Int"); def f(d: Double) = println("Double"); def f(a: Any) = println("Any"); }
defined module A

scala> A.f(5)
Int

scala> A.f("foo")
Any

scala> def g[@specialized(Int, Double) T](x: T) = A.f(x)
g: [T](x: T)Unit

scala> g(5)
Any
```